tests/Paginator/KeysetPaginatorTest.php

```diff
@@ -1168,4 +1168,84 @@ public function testGetPageToken(): void
 
         $this->assertSame($token, $paginator->getToken());
     }
+
+    public function testNextPage(): void
+    {
+        $dataSet = [['id' => 1], ['id' => 2], ['id' => 3]];
+        $sort = Sort::only(['id'])->withOrderString('id');
+        $dataReader = (new IterableDataReader($dataSet))->withSort($sort);
+        $paginator = (new KeysetPaginator($dataReader))->withPageSize(2);
+
+        // Test first page has next page
+        $nextPageReader = $paginator->nextPage();
+        $this->assertInstanceOf(KeysetPaginator::class, $nextPageReader);
+
+        // Verify the next page returns correct data
+        $nextPageData = array_values($this->iterableToArray($nextPageReader->read()));
+        $this->assertSame([['id' => 3]], $nextPageData);
+
+        // Test that returns null when there are no more pages
+        $this->assertNull($nextPageReader->nextPage());
+    }
+
+    public function testNextPageIterativeReading(): void
+    {
+        $dataSet = [['id' => 1], ['id' => 2], ['id' => 3]];
+        $sort = Sort::only(['id'])->withOrderString('id');
```

Comment:
```suggestion
        $sort = Sort::only(['id']);
```

Should work without `withOrderString('id')`.